examples/cookbook/recipe_plot_allocations.cpp

```diff
@@ -0,0 +1,57 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2018-2019, Lawrence Livermore National Security, LLC.
+// Produced at the Lawrence Livermore National Laboratory
+//
+// Created by David Beckingsale, david@llnl.gov
+// LLNL-CODE-747640
+//
+// All rights reserved.
+//
+// This file is part of Umpire.
+//
+// For details, see https://github.com/LLNL/Umpire
+// Please also see the LICENSE file for MIT license.
+//////////////////////////////////////////////////////////////////////////////
+#include "umpire/Umpire.hpp"
+#include "umpire/Allocator.hpp"
+#include "umpire/ResourceManager.hpp"
+#include "umpire/strategy/DynamicPool.hpp"
+
+#include <fstream>
+
+int main(int, char**) {
+  auto& rm = umpire::ResourceManager::getInstance();
+
+  auto allocator = rm.getAllocator("HOST");
+  auto pooled_allocator = rm.makeAllocator<umpire::strategy::DynamicPool>("HOST_POOL",
+                                                                          allocator,
+                                                                          1024 * 16);
+
+  void* a[4];
+  for (int i = 0; i < 4; ++i) a[i] = pooled_allocator.allocate(1024);
+
+  // Create fragmentation
+  pooled_allocator.deallocate(a[2]);
+  a[2] = pooled_allocator.allocate(1024 * 2);
+
+  // Output the records from the underlying host allocator
+  {
+    std::ofstream out("allocator.log");
+    umpire::print_allocator_records(allocator, out);
+    out.close();
+  }
+
+  // Output the records from the pooled allocator
+  {
+    std::ofstream out("pooled_allocator.log");
+    umpire::print_allocator_records(pooled_allocator, out);
+    out.close();
+  }
+
+  for (int i = 0; i < 4; ++i) pooled_allocator.deallocate(a[i]);
+
+  // Visualize this using the python script. Example usage:
+  // tools/plot_allocations allocator.log gray 0.2 pooled_allocator.log purple 0.8
```

Comment:
This is a hold-over from calling the method `print_allocation_map`. Now that this is called `print_allocator_records`, should the cookbook be called `plot_allocator_records`?